image_ipm/test/test_ipm.py

```diff
@@ -0,0 +1,196 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Tuple
+
+from cv_bridge import CvBridge
+from geometry_msgs.msg import TransformStamped
+from image_ipm.ipm import ImageIPM
+import numpy as np
+from numpy.lib import recfunctions as rfn
+import rclpy
+from rclpy.node import Node
+from rclpy.parameter import Parameter
+from sensor_msgs.msg import CameraInfo, Image, PointCloud2
+from sensor_msgs_py.point_cloud2 import read_points, read_points_numpy
+from std_msgs.msg import Header
+from tf2_msgs.msg import TFMessage
+
+cv_bridge = CvBridge()
+
+# Dummy CameraInfo Message
+camera_info = CameraInfo(
+        header=Header(
+            frame_id='camera_optical_frame',
+        ),
+        width=2048,
+        height=1536,
+        binning_x=4,
+        binning_y=4,
+        k=[1338.64532, 0., 1024.0, 0., 1337.89746, 768.0, 0., 0., 1.])
+
+img_center_x = camera_info.width / camera_info.binning_x // 2
+img_center_y = camera_info.height / camera_info.binning_y // 2
+
+
+def standard_ipm_image_test_case(
+        input_topic: str,
+        input_msg: Image,
+        output_topic: str,
+        mode: str = 'mask') -> Tuple[PointCloud2, Image]:
+    # Init ros
+    rclpy.init()
+    # Create IPM node
+    node = ImageIPM()
+    # Create test node which comunicates with the IPM node
+    test_node = Node('test_handler')
+    # Create publishers to send data to the IPM node
+    ball_pub = test_node.create_publisher(
+        Image, input_topic, 10)
+    camera_info_pub = test_node.create_publisher(
+        CameraInfo, 'camera_info', 10)
+    tf_pub = test_node.create_publisher(
+        TFMessage, 'tf', 10)
+
+    # Create a shared reference to the recived message in the local scope
+    received_msg: List[Optional[PointCloud2]] = [None]
+
+    # Create a callback with sets this reference
+    def callback(msg):
+        received_msg[0] = msg
+
+    # Subscribe to IPM results
+    test_node.create_subscription(
+        PointCloud2, output_topic, callback, 10)
+
+    # Create header message for the current time stamp in the camera frame
+    header = Header(
+        stamp=node.get_clock().now().to_msg(),
+        frame_id='camera_optical_frame')
+
+    # Create a dummy transform from the camera to the base_footprint frame
+    tf = TransformStamped(
+        header=header,
+        child_frame_id='base_footprint',
+    )
+    tf.transform.translation.z = 1.0
+    tf.transform.rotation.x = 0.0
+    tf.transform.rotation.w = 1.0
+
+    # Publish the dummy transform
+    tf_pub.publish(TFMessage(
+        transforms=[tf]
+    ))
+    # Spin the ipm to process the new data
+    rclpy.spin_once(node, timeout_sec=0.1)
+
+    node.set_parameters([Parameter("type", value=mode)])
```

Comment:
```suggestion
    node.set_parameters([Parameter('type', value=mode)])
```
This throws off the CI.